internal/util/vault.go

```diff
@@ -136,6 +149,35 @@ func InitVaultKMS(kmsID string, config, secrets map[string]string) (EncryptionKM
 	return kms, nil
 }
 
+func parseVaultToken(kms *VaultKMS, config, secrets map[string]string) error {
+	var ok bool
+	var err error
+	var token, authToken, vaultTokenFromSecret string
+	authToken, ok = config["vaultTokenAuth"]
+	if ok {
+		kms.VaultTokenAuth, err = strconv.ParseBool(authToken)
+		if err != nil {
+			return fmt.Errorf("failed to parse 'vaultTokenAuth' for vault  kms config: %v", err)
```

Comment:
when returning errors, use `%w` in the formatting string to be able to use `errors.Is()` and similar